src/ast/utils/isReservedWord.ts

```diff
@@ -0,0 +1,26 @@
+const es3ReservedWordsMap: { [word: string]: boolean } = `
+		abstract boolean break byte case catch char class
+		const continue debugger default delete do double else
+		enum export extends false final finally float for
+		function goto if implements import in instanceof int
+		interface long native new null package private
+		protected public return short static super switch synchronized
+		this throw throws transient true try typeof var
+		void volatile while with
+	`
+	.trim()
+	.split(/\s+/)
+	.reduce((acc: { [word: string]: boolean }, word: string) => {
+		acc[word] = true;
+		return acc;
+	}, {});
+
+export function isReservedWord(word: string, ecmaVersion: number) {
+	switch (ecmaVersion) {
+		case 3:
+			return !!es3ReservedWordsMap[word];
+
+		default:
+			throw new Error('Not implemented');
+	}
+}
```

Comment:
Two questions: Why not just use the `RESERVED_NAMES` in src/utils/reservedNames.ts? And do the Java names in this list such as `byte` and `int` really cause issues on ES3 systems? If so, adding them to the RESERVED_NAMES would be ok for me, otherwise we might just omit them as on all systems I am aware of, they do not cause issues.

Reply:
Also I do not think there is any reason to distinguish ECMA versions here, just use a single map that covers all versions (ideally RESERVED_NAMES).

Reply:
I guess `rollup` has performance as one of key characteristics, so RESERVED_NAMES is converted to hash map to have O(1) search complexity. Function is provided to hide implementation details for this variable.

Will rename to `RESERVED_NAMES` as recommended.

Will remove ECMA version as recommended.

Tested all keywords on IE6 and IE7 and you are right, not all of them need to be quoted. Will exclude not needed keywords.

Reply:
What I mean to say, I think there already exists a complete table of reserved words in Rollup's code base in src/utils/reservedNames.ts that should be reused